### PR TITLE
Use threadpools and other optimizations for inferencing code

### PR DESCRIPTION
**Pull request by dkirby-ms**

### CODE REVIEW

File: agora/footfall_ai_api/src/video_capture.py [hunk 1]
```diff
@@ -35,17 +34,19 @@ def _reader(self):
                 if not self.running:
                     return
             
-           # Save the frame if it is at the specified interval
+            # Save the frame if it is at the specified interval
             if self.frame_count % self.frame_interval == 0:
                 self.q.put(cv2.resize(frame, (640, 360)))
```
Comment (fcabrera23) [inline agora/footfall_ai_api/src/video_capture.py:39]: What about having two env variables, FRAME_HEIGH and FRAME_WIDTH and just resize it here to the desired size? Maybe better than doing two resigning, first to 640 *320 and then the other one you do as part of processor

File: agora/footfall_ai_api/src/video_processor.py [hunk 2]
```diff
@@ -73,28 +77,38 @@ def start(self):
...
+        # Clear the queue to free up memory
+        with self.processed_frame_queue.mutex:
+            self.processed_frame_queue.queue.clear()
         print(f"Stopped processing thread for video {self.index}")
 
     def process_frames(self):
         processing_times = collections.deque(maxlen=200)
         frame_count = 0
         last_time = time.time()
+        original_width, original_height = 640, 360  # Original resolution
```
Comment (fcabrera23) [inline agora/footfall_ai_api/src/video_processor.py:98]: See my comment above, small detail

File: agora/footfall_ai_api/src/video_processor.py [hunk 3]
```diff
@@ -73,28 +77,38 @@ def start(self):
...
+        scaling_factor_y = new_height / original_height
+
         while self.running:
             frame, success = self.vs.read()
-            original_frame = frame.copy()
             if not success:
                 continue
 
+            # Reduce input resolution
+            frame = cv2.resize(frame, (new_width, new_height))
```
Comment (fcabrera23) [inline agora/footfall_ai_api/src/video_processor.py:109]: What about the new frame size? Does the model detects people with such a small size?

File: agora/shopper_insights_api/src/app.py [hunk 4]
```diff
@@ -68,7 +71,7 @@ def video_feed():
         return "Invalid JSON format", 400
     
     camera_name = data["cameraName"]       
-    return Response(generate(data, camera_name), mimetype='multipart/x-mixed-replace; boundary=frame')
+    return Response(executor.submit(generate, data, camera_name).result(), mimetype='multipart/x-mixed-replace; boundary=frame')
```
Comment (fcabrera23) [inline agora/shopper_insights_api/src/app.py:74]: If i understand this correctly, every time someone calls the video_feed, you create a new thread, even if it's the same camera. But then the camera processor and captures are the always the same 2 threads for a specific camera - is that correct?

**Review by fcabrera23 — COMMENTED** [review]
@dkirby-ms please check my comments

File: agora/shopper_insights_api/src/app.py (hunk 4 above)
Comment (dkirby-ms) [reply agora/shopper_insights_api/src/app.py:74]: agreed, we dont need to pool the api calls themselves, just the video processor. copilot gonna copilot

File: agora/footfall_ai_api/src/video_processor.py (hunk 3 above)
Comment (dkirby-ms) [reply agora/footfall_ai_api/src/video_processor.py:109]: agree

**Pull request closed without merging**